Deprecated function

Muhammad-Rebaal opened the pull request:
Added the functional change [#6758](https://github.com/aiidateam/aiida-core/pull/6758#event-16322583852)

unkcpz commented:
It is an empty PR, could you check you push the change?

Muhammad-Rebaal commented:
> It is an empty PR, could you check you push the change?

I've Updated

danielhollas commented on src/aiida/common/utils.py:
```diff
@@ -446,6 +446,11 @@ def join_labels(labels, join_symbol='|', threshold=1.0e-6):
     return new_labels
 
 
+<<<<<<< HEAD
+
+
+=======
+>>>>>>> 64dd239b1e5c05153823b2eda1fb3ca3f20b2436
```
```suggestion
```

danielhollas commented on src/aiida/orm/utils/node.py:
```diff
@@ -60,8 +60,8 @@ def load_node_class(type_string):
     if base_path.startswith('process'):
-        entry_point_name = base_path.removeprefix('data.')
-        return load_entry_point('aiida.node', entry_point_name)
+        entry_point_name = base_path.removeprefix('nodes.')
+        return load_entry_point('aiida.node', entry_point_name) 
```
I believe the original code was incorrect

```suggestion
        entry_point_name = base_path.removeprefix('process.')
        return load_entry_point('aiida.node', entry_point_name) 
```

danielhollas commented on src/aiida/orm/utils/node.py:
```diff
@@ -60,8 +60,8 @@ def load_node_class(type_string):
             return Data
 
     if base_path.startswith('process'):
```
Based on the similar code above, I think this should be:

```suggestion
    if base_path.startswith('process.'):
```

danielhollas reviewed (changes requested):
@Muhammad-Rebaal thank you for the PR, see my review comments.

We need to add tests covering this code. Can you please try to add them to `tests/orm/utils/test_node.py`. You should be able to run the tests locally by 

```console
pytest tests/orm/utils/test_node.py
```

Let us know if you need asssistance with them:

khsrali commented:
> Added the functional change [#6758](https://github.com/aiidateam/aiida-core/pull/6758#event-16322583852)

If it's possible please give more detailed explanation.

Also the title: "Deprecated function" is not descriptive at all, please revise.

Muhammad-Rebaal commented:
> @Muhammad-Rebaal thank you for the PR, see my review comments.
> 
> We need to add tests covering this code. Can you please try to add them to `tests/orm/utils/test_node.py`. You should be able to run the tests locally by
> 
> ```
> pytest tests/orm/utils/test_node.py
> ```
> 
> Let us know if you need asssistance with them:

Yeah sure @danielhollas ! But if you describe what is your intention by adding a test in that particular file about those functions that've changed then it would be easier for me to implement that.

Muhammad-Rebaal commented:
> > Added the functional change [#6758](https://github.com/aiidateam/aiida-core/pull/6758#event-16322583852)
> 
> If it's possible please give more detailed explanation.
> 
> Also the title: "Deprecated function" is not descriptive at all, please revise.

I've changed that and thank you for your assistance .

unkcpz commented:
Hi @Muhammad-Rebaal, please check the comment above, there are diff compares in the code not cleaned.

Muhammad-Rebaal commented:
> Hi @Muhammad-Rebaal, please check the comment above, there are diff compares in the code not cleaned.

Hi ! Yes , I've updated the suggested change in `src/aiida/orm/utils/node.py` file which is in line 62 and 63.

Muhammad-Rebaal commented:
Hey @danielhollas @unkcpz ! 
Need some assistance regarding the test part . By running this command : `pytest tests/orm/utils/test_node.py`
I ran into some issues like : 
`E           PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\Temp\\AppData\\Local\\Temp\\pytest-of-Temp\\pytest-22\\28efecaa984d0d6c847fc2b33421ae0c0\\.aiida\\tmpgopjmudt' -> 'C:\\Users\\Temp\\AppData\\Local\\Temp\\pytest-of-Temp\\pytest-22\\28efecaa984d0d6c847fc2b33421ae0c0\\.aiida\\config.json'

C:\Users\Temp\AppData\Roaming\Python\Python312\site-packages\aiida\manage\configuration\config.py:782: PermissionError
========================= slowest 50 durations ========================= 

(2 durations < 1s hidden.  Use -vv to show these durations.)
======================= short test summary info ======================== 
ERROR tests/orm/utils/test_node.py::test_load_node_class_fallback - PermissionError: [WinError 32] The process cannot access the file be...  `

If you know how to resolve this please let me know . I've done the previous parts of removing the conflict and adding the suggested changes .

danielhollas commented:
I am not a Windows user, but it looks like perhaps you have the file opened in an editor or somewhere else and pytest cannot access it?

Muhammad-Rebaal commented:
Hey @danielhollas @unkcpz !
I've added the tests in the `tests/orm/utils/test_node.py` directory and implemented the tests that are passed here's the picture :
 
<img width="455" alt="image" src="https://github.com/user-attachments/assets/8873e99b-32a4-4504-affb-68346df4d2c7" />

And a little request if you could assign me those issues that I've resolved [#6760](https://github.com/aiidateam/aiida-core/pull/6760) , [#6758](https://github.com/aiidateam/aiida-core/pull/6758) , [#6705](https://github.com/aiidateam/aiida-core/issues/6705) as a token of appreciation for my contributions.

danielhollas commented on tests/orm/utils/test_node.py:
```diff
@@ -23,3 +23,48 @@ def test_load_node_class_fallback():
+    # Test node prefix removal
+    with pytest.warns(UserWarning):
+        loaded_class = load_node_class('node.data.dict.')
+        assert loaded_class == Data
```
Let's put the asserts outside of the context manager (same below)
```suggestion
    assert loaded_class == Data
```

Muhammad-Rebaal replied:
I think this would cause a syntax error

danielhollas replied:
Why would it cause syntax error?

Muhammad-Rebaal replied:
Apologies ! I didn't see that line of the code I thought the indentation problem but it isn't there

danielhollas commented on src/aiida/orm/utils/node.py:
```diff
@@ -59,8 +59,8 @@ def load_node_class(type_string):
-    if base_path.startswith('process'):
-        entry_point_name = base_path.removeprefix('data.')
+    if base_path.startswith('process.'):
+        entry_point_name = base_path.removeprefix('process.')
```
Can you try removing this line to see if that fixes the failing tests?

```suggestion
```

You can run a subset of test suite with `pytest -m presto -n auto tests/`

Muhammad-Rebaal replied:
Hey @danielhollas  ! I think the command you suggest to run in the terminal  is not correct because it would run all the tests that would be in the `tests` directory and because of all those tests I'm getting errors rather it should be `pytest -m presto -n auto tests/orm/utils/test_node.py` because it would check only the written test in the test_node.py file which was written by which I'm getting tests are passed . Here's the image : 

<img width="900" alt="image" src="https://github.com/user-attachments/assets/42706f1f-894f-4cc6-b87e-2577ffcfa498" />


If I'm missing something please let me know .
Thank You!

Muhammad-Rebaal replied:
> > I think the command you suggest to run in the terminal is not correct because it would run all the tests that would be in the tests directory and because of all those tests I'm getting errors
> 
> Yes, but that is the point. Clearly the change in this PR is breaking some of those pre-existing tests, as you can see here on GitHub. So we need to figure out why those tests are failing.

ok Now I understand ! If you've done this part of tracking those tests then let me know , so the code would be refactored faster else I'll try to track those failing tests and fix them for good 👍🏻

danielhollas reviewed (changes requested):
Hmm, there are a lot of failing tests so I suspect the code change here is actually not correct.

Muhammad-Rebaal commented:
> Hmm, there are a lot of failing tests so I suspect the code change here is actually not correct.

Yeah sure! I saw that  , those failing tests . I'll resolve them as soon as possible.

danielhollas commented on src/aiida/orm/utils/node.py:
```diff
@@ -59,8 +59,8 @@ def load_node_class(type_string):
-    if base_path.startswith('process'):
-        entry_point_name = base_path.removeprefix('data.')
+    if base_path.startswith('process.'):
+        entry_point_name = base_path.removeprefix('process.')
         return load_entry_point('aiida.node', entry_point_name)
```
Based on my local testing, I believe the correct code is this one
```suggestion
        return load_entry_point('aiida.node', base_path)
```

danielhollas commented on src/aiida/orm/utils/node.py:
```diff
@@ -59,9 +59,8 @@ def load_node_class(type_string):
-    if base_path.startswith('process'):
-        entry_point_name = base_path.removeprefix('data.')
-        return load_entry_point('aiida.node', entry_point_name)
+    if base_path.startswith('process.'):
```
```suggestion
    if base_path.startswith('process'):
```

Muhammad-Rebaal commented:
Hey @danielhollas !
Hope you are fine !

Is this issue resolved ? Or needs to have some work done ? Please let me know ?

Thank You !

danielhollas commented on tests/orm/utils/test_node.py:
```diff
@@ -23,3 +23,48 @@ def test_load_node_class_fallback():
+def test_load_node_class_with_node_prefix():
+    """Test the behavior of load_node_class with node prefix."""
+    # Test node prefix removal
+    with pytest.warns(UserWarning):
```
Can you please match the specific warning, the same way you do in `test_load_node_class_with_data_prefix`? Same on line 39 below

danielhollas commented on tests/orm/utils/test_node.py:
```diff
@@ -23,3 +23,48 @@ def test_load_node_class_fallback():
     with pytest.warns(UserWarning):
         loaded_class = load_node_class('__main__.SubData.')
     assert loaded_class == Data
+
+
+# Test for Remove_prefix
```
```suggestion
```

danielhollas commented on tests/orm/utils/test_node.py:
```diff
@@ -23,3 +23,48 @@ def test_load_node_class_fallback():
+
+    # Test process prefix with empty subtype - should return ProcessNode
+    loaded_class = load_node_class('process.')
+    assert (
+        loaded_class == ProcessNode
+    )  # Changed from Data to ProcessNode because that's what the implementation returns
```
Let's remove this comment, it's more misleading than informative imo
```suggestion
    assert loaded_class == ProcessNode
```

danielhollas reviewed (changes requested):
@Muhammad-Rebaal thanks for the ping! Regarding the changes themselves, I just have some minor cleanup notes.

It would be great if we could get 100% test coverage for the `orm/utils/node.py` file in this PR, since you've added a bunch of tests already.
If you're up for it, you can check which branches are not tested on Codecov report here:
https://app.codecov.io/gh/aiidateam/aiida-core/pull/6760/blob/src/aiida/orm/utils/node.py

![image](https://github.com/user-attachments/assets/418c37bf-9d45-4653-ac2e-12cacea3f0d2)

There's actually not much missing!

Muhammad-Rebaal commented:
> @Muhammad-Rebaal thanks for the ping! Regarding the changes themselves, I just have some minor cleanup notes.
> 
> It would be great if we could get 100% test coverage for the `orm/utils/node.py` file in this PR, since you've added a bunch of tests already. If you're up for it, you can check which branches are not tested on Codecov report here: https://app.codecov.io/gh/aiidateam/aiida-core/pull/6760/blob/src/aiida/orm/utils/node.py
> 
> ![image](https://private-user-images.githubusercontent.com/9539441/426081482-418c37bf-9d45-4653-ac2e-12cacea3f0d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5MzM0MjIsIm5iZiI6MTc0MjkzMzEyMiwicGF0aCI6Ii85NTM5NDQxLzQyNjA4MTQ4Mi00MThjMzdiZi05ZDQ1LTQ2NTMtYWMyZS0xMmNhY2VhM2YwZDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjVUMjAwNTIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTYxOGExYWY5NjAyYTBlNzVjZDc5Y2JkMjdiOTE3NWVjMTY4ZmQ5Nzg3MTRhZTA0YTgwMjY0NWM3NWIzZDVlNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.s7n66lvv-pQVNwXnSCKv6IG2LMwfZgcsoaVsuzyhLDM)
> 
> There's actually not much missing!

Yeah sure ! It would be a pleasure to do that .

Muhammad-Rebaal commented:
Hi @danielhollas , @unkcpz  !

Hope you are fine !

Could you please review code . I've added the 100% test coverage for `node.py` in `test_node.py` . I've also tested it locally by `coverage` a python library. 

Thank You !

danielhollas reviewed (changes requested):
@Muhammad-Rebaal thank you for adding all the tests! The file is now 100% covered. :tada: 

Unfortunately there are now extra changes  in `src/aiida/orm/utils/node.py` which makes it harder to review. Can we please focus only at the original issue with the superfluous `removeprefix`? Other changes can be done in a follow-up PR.

(I haven't gone through the tests themselves yet, will do that in the next iteration).

Thanks for your patience here, and apologies for late review.
